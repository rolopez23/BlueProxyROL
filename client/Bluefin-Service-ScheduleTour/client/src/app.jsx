/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import axios from 'axios';
import getDates from './helpers/getDates';

import {
  AppWrapper, Header, Paragraph, GlobalStyles, CalendarWrapper,
} from './styles/app-style';

import CalendarContainer from './CalendarContainer';
import ScheduleButton from './ScheduleButton';
import Contact from './Contact';
import Refund from './Refund';
import StartOffer from './StartOffer';
import LiveChat from './LiveChat';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      houseData: null,
      selectedDate: null,
      chat: false,
    };
    this.calendarEntryClick = this.calendarEntryClick.bind(this);
    this.scheduleClick = this.scheduleClick.bind(this);
    this.askQClick = this.askQClick.bind(this);
  }

  componentDidMount() {
    const { listingId } = this.props;
    if (listingId) {
      axios.get(`/house?listingId=${listingId}`)
        .then((response) => {
          this.setState({
            houseData: response.data[0],
          });
        });
    }
  }

  calendarEntryClick(data) {
    this.setState({
      selectedDate: data,
    });
  }

  scheduleClick() {
    // placeholder for scheduling data ...
    const { selectedDate } = this.state;
    // ...

    axios.post('/schedule', { selectedDate })
      .then((response) => {
        // eslint-disable-next-line no-console
        console.log(response.data);
      });
  }

  askQClick() {
    const { chat } = this.state;
    this.setState({ chat: !chat });
  }

  render() {
    const { houseData, selectedDate, chat } = this.state;

    return (
      <AppWrapper>
        <GlobalStyles />
        <Header>Go Tour This Home</Header>
        <CalendarWrapper>
          {
            houseData === null
              ? <p>Hello</p>
              : (
                <CalendarContainer
                  dates={getDates()}
                  clickFn={this.calendarEntryClick}
                  selectedDate={selectedDate}
                />
              )
          }
        </CalendarWrapper>
        <ScheduleButton
          clickFn={this.scheduleClick}
        />
        <Paragraph>It&apos;s free, with no obligation - cancel anytime</Paragraph>
        <Contact
          phone={houseData ? houseData.phone_number : 'Didnt get it'}
          clickFn={this.askQClick}
        />
        <Refund
          price={houseData
            ? houseData.listing_price : 'Didnt get it'}
        />
        <StartOffer />
        { chat ? <LiveChat /> : <></>}
      </AppWrapper>
    );
  }
}
