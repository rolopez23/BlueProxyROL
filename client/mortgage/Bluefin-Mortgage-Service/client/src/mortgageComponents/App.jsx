/* eslint-disable import/extensions */
import React from 'react';
import Styled from 'styled-components';
import axios from 'axios';
// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from 'prop-types';
// Component Imports
import AdList from './ads/AdList.jsx';
import PaymentCalculator from './paymentCalculator/PaymentCalculator.jsx';
// Model imports
import Mortgage from '../../mortgageModel/mortgage.js';

const AppWrapper = Styled.div`
  border-top: 1px solid #e2e2e2;
  display: block;
  position: relative;
  margin-top: 3rem;
  min-width: 600px;
  width: 728px;
  maxwidth: 728px;
`;


class App extends React.Component {
  constructor(props) {
    super(props);
    const { listingId } = this.props;
    this.state = {
      listingId,
      price: null,
      downPayment: null,
      rate: 3.5,
      ads: [],
      listing: {},
      payments: [],
      loanType: '30 Year Fixed',
      years: 30,
    };
    this.updatePrice = this.updatePrice.bind(this);
    this.updateMortgage = this.updateMortgage.bind(this);
    this.updateRate = this.updateRate.bind(this);
    this.updateLoanType = this.updateLoanType.bind(this);
  }

  componentDidMount() {
    const { listingId } = this.state;
    const url = `/listing${listingId}`;
    axios.get(url)
      .then((response) => {
        // console.log(response.data);
        const newAds = response.data.ads;
        const newListing = response.data.listing;
        const { price } = newListing;
        const downPayment = price * 0.2;
        this.updateMortgage(30, 3.5, price, downPayment);
        this.setState({
          ads: newAds,
          listing: newListing,
          price,
          downPayment,
        });
      });
  }

  // Section for helpers that update main App state in lower components

  // returns a payment train, creates a mortgage object but only saves payment
  // array to state
  // Inputs: Required: years, APR, price
  // Optional: downPayment, HOA, taxRate, homeownersInsurance, PMI
  updateMortgage(years, APR, price, downPayment, HOA, taxRate, homeownersInsurance, PMI) {
    const mortgage = new Mortgage(years, APR, price, downPayment, HOA, taxRate, homeownersInsurance, PMI);
    const payments = mortgage.getMonthlyPayment();
    // console.log(payments);
    this.setState({ payments });
    // console.log(this.state.payments);
  }

  updateLoanType(loanType) {
    let newYears = 30;
    const { rate, price, downPayment } = this.state;
    if (loanType === '15 Year Fixed') {
      newYears = 15;
    }
    this.setState({ loanType, years: newYears });
    this.updateMortgage(newYears, rate, price, downPayment);
  }

  updatePrice(price, downPayment) {
    const { rate, years } = this.state;
    this.setState({
      price,
      downPayment,
    });
    this.updateMortgage(years, rate, price, downPayment);
    // console.log('updated price', price, downPayment);
  }

  updateRate(rate) {
    const { years, price, downPayment } = this.state;
    this.setState({ rate });
    this.updateMortgage(years, rate, price, downPayment);
  }

  render() {
    const {
      listing, ads, price, downPayment, payments, loanType, rate,
    } = this.state;
    // console.log(price, downPayment);
    return (
      <AppWrapper>
        <PaymentCalculator
          rate={rate}
          payments={payments}
          loanType={loanType}
          price={price}
          downPayment={downPayment}
          updatePrice={this.updatePrice}
        />
        <AdList
          ads={ads}
          listing={listing}
          rateChange={this.updateRate}
          loanTypeChange={this.updateLoanType}
        />
      </AppWrapper>
    );
  }
}

App.propTypes = {
  listingId: PropTypes.number.isRequired,
};


export default App;
