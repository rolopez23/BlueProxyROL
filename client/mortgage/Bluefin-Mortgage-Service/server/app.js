//  import statements
const express = require('express');
const path = require('path');

// file dependencies
const db = require('../database/');
// declare file Path
const filePath = path.join(__dirname, '..', 'public');
// name App
const app = express();

app.use(express.static(filePath));

app.use(express.json());
app.use(express.urlencoded());

app.get('/listing:listingId', (req, res) => {
  // Get the id of the item, form req
  console.log('request', req.params);
  const { listingId } = req.params;
  const send = {};
  // Send request for that listing
  db.getListing(listingId)
    .then((data1) => {
      send.listing = data1;
      const { region } = data1;
      // console.log(region);
      return db.getMortgageAds(region);
    })
    .then((data2) => {
      send.ads = data2;
      // console.log(data, send.listing, send.ads);
      res.send(send);
    });
});

app.patch('/adClick', (req, res) => {
  // Get the id of the add
  console.log('recieved request', req.body);
  const adId = req.body.id;
  let adRecord = {};
  // console.log(adId);
  db.getAd(adId)
    .then((ad) => {
      // console.log('ad', ad.clicks);
      const newClick = ad.clicks + 1;
      adRecord = ad;
      adRecord.clicks = newClick;
      return db.updateAd(adId, newClick);
    })
    .then(() => {
      // res.location('./secureContact.html');
      // console.log(adRecord);
      res.send(JSON.stringify(adRecord));
      // res.redirect('./securecontanct.html');
      // console.log(res);
    });
  // Increment that add by one

  // Redirect to third party site.
});

module.exports = {
  app,
};
