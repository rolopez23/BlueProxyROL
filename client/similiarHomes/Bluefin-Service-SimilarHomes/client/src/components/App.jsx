import React from 'react';
import axios from 'axios';
import styled from 'styled-components';

// eslint-disable-next-line import/extensions
import SimilarListingsContainer from './SimilarListingsContainer.jsx';

class NearbyListingsContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      listings: [],
    };
  }

  componentDidMount() {
    const { listings } = this.state;
    if (!{ listings }.length) {
      this.getListings();
    }
  }

  getListings() {
    axios.get('http://localhost:4004/image')
      .then((listings) => {
        this.setState({
          listings: listings.data,
        });
      });
  }

  render() {
    const { listings } = this.state;
    const containers = listings.map((l) => <SimilarListingsContainer data={l} />);

    const ScrollingComponent = styled.div`
      display: grid;
      grid-gap: 16px;
      grid-template-columns: repeat(${listings.length}, 297px);
      grid-template-rows: 1fr;
      width: 616px;
      overflow-y: none;
      overflow-x: scroll;
      scroll-snap-type: inline mandatory;
    `;

    const ScrollingContainer = styled.div`
      height: 400px;
    `;

    const output = (
      <div id="section-title">
        <h2 style={{ 'font-family': '"Libre Franklin", sans-serif' }}>Nearby Similar Homes</h2>
        <ScrollingContainer>
          <ScrollingComponent className="similar-listings">
            {containers}
          </ScrollingComponent>
        </ScrollingContainer>
      </div>
    );
    return output;
  }
}

export default NearbyListingsContainer;
