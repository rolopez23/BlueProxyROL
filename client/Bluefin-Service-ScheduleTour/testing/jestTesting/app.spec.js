
/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import App from './app'
import { Counter, testFunction } from './app';

describe('App', () => {
  it('snapshot renders', () => {
    const app = renderer.create(<App />);
    let tree = app.toJSON();
    expect(app).toMatchSnapshot();
  });

  describe('Unit Test', () => {
    it('testFunction works', () => {
      let object = { animal : 'ox'};
      let moreProperties = { horns: true, color: 'brown' };
      var newObject = testFunction(object, moreProperties);
      expect(newObject).toEqual({ animal: 'ox', horns: true, color: 'brown' });
    })
  })
  describe('Counter', () => {
    test('snapshot renders', () => {
      const component = renderer.create(<Counter counter={1} />);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
})
