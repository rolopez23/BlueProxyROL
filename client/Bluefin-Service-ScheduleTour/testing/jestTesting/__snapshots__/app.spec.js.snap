// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Counter snapshot renders 1`] = `
<div>
  <p>
    1
  </p>
</div>
`;

exports[`App snapshot renders 1`] = `
<div>
  App me down!
  <div>
    <p>
      0
    </p>
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    increment
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    deccrement
  </button>
</div>
`;
